src/CaptureFileInfo/include/CaptureFileInfo/CaptureFileInfo.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CAPTURE_FILE_INFO_CAPTURE_FILE_INFO_H_
+#define CAPTURE_FILE_INFO_CAPTURE_FILE_INFO_H_
+
+#include <QDateTime>
+#include <QFileInfo>
+#include <filesystem>
+#include <utility>
+
+namespace orbit_capture_file_info {
+
+class CaptureFileInfo {
+ public:
+  explicit CaptureFileInfo(const QString& path)
+      : file_info_(path), last_used_(QDateTime::currentDateTime()) {}
+  explicit CaptureFileInfo(const QString& path, QDateTime last_used)
+      : file_info_(path), last_used_(std::move(last_used)) {}
+
+  [[nodiscard]] QString FilePath() const { return file_info_.filePath(); }
+  [[nodiscard]] QString FileName() const { return file_info_.fileName(); }
+
+  [[nodiscard]] QDateTime LastUsed() const { return last_used_; }
+  [[nodiscard]] QDateTime Created() const { return file_info_.birthTime(); }
+
+  [[nodiscard]] bool FileExists() const { return file_info_.exists() && file_info_.isFile(); }
+
+  void Touch() { last_used_ = QDateTime::currentDateTime(); }
+
+ private:
+  QFileInfo file_info_;
```

Comment:
out of curiosity - how does it handle io errors?

Reply:
I dont know. I googled and I did not find a lot of information. I assume when a call to (for example) `.exists()` fails it just returns false. In this case maybe `QFileDevice::error()` [link](https://doc.qt.io/qt-5/qfiledevice.html#error) returns the error code why it failed. This would be corresponding to the `std::filesystem::exists(path, error_code)` call. But this is only my best guess, I dont know for sure.